### DISCUSSION TITLE
Shared experience / spatial anchors

### DISCUSSION THREAD

**Discussion by llfab** [StereoKit/StereoKit#317] [Q&A] [answered]
As far as I understand there is currently no support for shared experiences for 2 Hololenses, right. I am talking about Hololens 2. I believe something like local anchors or Azure Spatial anchors would be needed for that. Correct?

**Comment by maluoi** [comment, marked as answer]
You totally can! :)

For co-location, there's a few tools open to you:
- Azure Spatial Anchors does work with StereoKit, I don't personally have a _ton_ of experience with this, but @brunoshine built a [sample StereoKit+ASA project](https://github.com/brunoshine/StereoKit.Samples.AzureSpatialAnchors) for reference.
- [QR codes](https://stereokit.net/Pages/Guides/Using-QR-Codes.html) are always fast and easy.
- [Regular world anchors](https://github.com/maluoi/StereoKit/blob/master/Examples/StereoKitTest/Demos/DemoWorldAnchor.cs) may be shareable over the network? I feel like I've seen that in some Microsoft docs somewhere.
- The "Two Point" method. If all users touch the same two points (like the corners of a table), it's quite straightforward to turn that into a common pose.

For networking, basically anything could work! There's a lot of stuff out there on NuGet and the .Net community. Here's a few I've seen people using with StereoKit:
- [Ruffles](https://github.com/MidLevel/Ruffles)
- [LiteNet](https://github.com/RevenantX/LiteNetLib)
- Stuff like [Photon](https://www.photonengine.com/) may come from Unity originally, but there's nothing preventing it from being used here as well.

**Reply by llfab** [reply]
Great. For networking, we currently use http/REST which is great for unidirectional networking.
